codec/types/interface_registry.go

```diff
@@ -147,6 +149,12 @@ func (registry *interfaceRegistry) UnpackAny(any *Any, iface interface{}) error
 		return fmt.Errorf("can't proto unmarshal %T", msg)
 	}
 
+	// Firstly check that we don't have extraneous and
+	// mistyped/mismatched fields in the bytes to be unmarshaled.
+	if err := enforceproto.CheckMismatchedProtoFields(any.Value, msg); err != nil {
+		return err
+	}
+
```

Comment:
This is a logical place to put this, but I'd rather leave it out of this method. The reason being that unknown field rejection should only happen when processing an untrusted message in `CheckTx`. `UnpackAny` is used in other scenarios such as reading data from the state store where we don't care to check for unknown fields. So let's deal with the `Any` case elsewhere.

```suggestion
```

Reply:
Sounds good thank you. Sure, I can move it elsewhere.

Reply:
I've removed it for now as it'll be trivial to place once review of the core functionality is done.